### PR TITLE
[feat]: [PLG-3019]: NG Audits for ModuleLicense collection

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes

**Scenarios tested**:

**Case 1**: display of ModuleLicense audits on audits page
Screenshot 2023-11-27 at 7 38 05 PM

**********
**Case 2**: created CD ModuleLicense
Screenshot 2023-11-27 at 7 38 13 PM

************
**Case 3**:  updated CD ModuleLicense
Screenshot 2023-11-28 at 3 30 53 AM

*************
**Case 4**: deleted CD module license
Screenshot 2023-11-28 at 3 38 21 AM

**************
**Case 5**: updated CI module license
Screenshot 2023-11-28 at 3 43 58 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56007)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger messagecheck

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger ut4

File: 930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.licensing;
+
+public class ModuleLicenseConstants {
+  public static final String RESOURCE_NAME = "Module License";
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java:11]: nit: should we rename this as `Module_License` instead so it becomes one word ?

**Review by neelam-harness — APPROVED** [review]
LGTM overall

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56007)* status: 18 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java` line 11 at r5](https://reviewable.io/reviews/harness/harness-core/56007#gh-1407851025:-NkL_v5W6fJglSeKLXQC:bg2ayw3) ([raw file](https://github.com/harness/harness-core/blob/a8172ade53c1c4d2aa320a651f4a97ba790f7abe/930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java#L11)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

nit: should we rename this as `Module_License` instead so it becomes one word ?
</blockquote></details>

this string is displayed on Harness Audit trails UI so has to readable and not one word
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java` line 12 at r5](https://reviewable.io/reviews/harness/harness-core/56007#-NkL8abt2Pltkfj5p-Od:-NkLacfY1-JuTsu-JR1Z:b-lqigd9) ([raw file](https://github.com/harness/harness-core/blob/a8172ade53c1c4d2aa320a651f4a97ba790f7abe/930-ng-license-manager/src/main/java/io/harness/licensing/ModuleLicenseConstants.java#L12)):*
<details><summary><i>Previously, Xin-Harness (Xin Shao) wrote…</i></summary><blockquote>

nit: should be moduleLicenseCreated
In addition, we should reuse our constant class NGLicensingEntityConstants or combine these two together.
</blockquote></details>

1st comment: ModuleLicenseCreated is not a variable hence need not be in camel case, its used as variant of Audit event in outbox handler.

2nd comment: yes we can do but NGLicensingEntityConstants does not seem to be widely used there are only 2 variables inside it. For now its good to have a different constants class dedicated for audits.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/event/ModuleLicenseEvent.java` line 43 at r5](https://reviewable.io/reviews/harness/harness-core/56007#-NkLA1TC7dLLo65aRNZu:-NkLbH5qEE2gU1emY6zX:b-nwxwcx) ([raw file](https://github.com/harness/harness-core/blob/a8172ade53c1c4d2aa320a651f4a97ba790f7abe/930-ng-license-manager/src/main/java/io/harness/licensing/event/ModuleLicenseEvent.java#L43)):*
<details><summary><i>Previously, Xin-Harness (Xin Shao) wrote…</i></summary><blockquote>

It look a bite strange to me to create this new hash map inside this method. Can this map create outside the method?
</blockquote></details>

In order to build labels for Resource object we need to instantiate a HashMap, so it's rightly placed here. 
If you look at other implementations of Event there also its used in this manner.

**Review by mohitg0795 — APPROVED** [review]

**Review by Xin-Harness — APPROVED** [review]

**Review by Xin-Harness — APPROVED** [review]
> Reviewed 1 of 12 files at r1, 1 of 5 files at r2, 1 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56007)* status: 18 file(s) to review. Waiting on team(s): del-team.

**Review by Xin-Harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56007)* status: 18 file(s) to review. Waiting on team(s): del-team.

**Comment by kapil-harness** [comment]
trigger review

**Review by AnupamIO — APPROVED** [review]

**Comment by akshay-khole** [comment]
@kapil-harness @neelam-harness  this change has broken licensing on smp 
https://harness.atlassian.net/browse/SMP-2379
was this tested against any smp set up ?

**Pull request merged**